### PR TITLE
TBD-13: Support custom copy-to-clipboard form fields

### PR DESCRIPTION
**Pull request by wazelin**
- TBD-13 feat: support `copy-to-clipboard` action on any form field with `copy-to-clipboard` class

Required by [taoLti extension](https://github.com/oat-sa/extension-tao-lti/pull/232).

### CODE REVIEW

File: views/js/form/post-render-props.js [hunk 1]
```diff
@@ -23,61 +23,87 @@ define([
...
-                        feedback().error(__('Your browser does not support copying to the clipboard'));
-                    }
-                }),
-            $fieldBox = $('<span>', {class: 'uri-container'});
+            $fields = $container.find('#id, .copy-to-clipboard');
 
         // if the field has already been added
         if ($('.uri-container').length) {
             return;
         }
```
Comment (habemuscode) [inline views/js/form/post-render-props.js]: if ($('.uri-container').length) return;

Short-form to try to follow code climate rule of 25 lines per function

File: views/js/form/post-render-props.js [hunk 2]
```diff
@@ -23,61 +23,87 @@ define([
...
 
-        if (!isInstanceForm || !$idField.length) {
+        if (!isInstanceForm) {
             return;
         }
```
Comment (habemuscode) [inline views/js/form/post-render-props.js]: Short-form to try to follow code climate rule of 25 lines per function

File: views/js/form/post-render-props.js [hunk 3]
```diff
@@ -23,61 +23,87 @@ define([
...
-            .after($('<div>')
-                .append([$label, $fieldBox]));
-        $fieldBox.height($copyField.outerHeight());
-
-        $copyField.addClass('final')
-        // re-add value
-            .val(value);
+        if (!$fields.length) {
+            return;
+        }
```
Comment (habemuscode) [inline views/js/form/post-render-props.js]: Short-form to try to follow code climate rule of 25 lines per function

File: views/js/form/post-render-props.js [hunk 4]
```diff
@@ -23,61 +23,87 @@ define([
...
+            } else {
+                $fieldBox.height($field.outerHeight());
+                $field.wrap($fieldBox);
+                $field.parent().append($button);
+            }
+
+            $button.on('click', createCopyToClipboardHandler($field));
+            $field.addClass('final')
+                // re-add value
+                .val(value);
```
Comment (habemuscode) [inline views/js/form/post-render-props.js]: $field.addClass('final').val(value); For 1 chain, we can put in same line

File: views/js/form/post-render-props.js (hunk 3 above)
Comment (wazelin) [reply views/js/form/post-render-props.js]: I absolutely disagree with that approach and would like to discuss it on our retro.

**Pull request merged**